src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -786,6 +787,23 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
                 case NI_VectorT128_Sum:
                 {
                     GenTree* tmp;
+
+                    switch (simdBaseType)
+                    {
+                        case TYP_BYTE:
+                        case TYP_UBYTE:
+                        case TYP_SHORT:
+                        case TYP_USHORT:
+                        case TYP_INT:
+                        case TYP_UINT:
+                        {
+                            tmp = gtNewSimdAsHWIntrinsicNode(simdType, op1, NI_AdvSimd_Arm64_AddAcross, simdBaseJitType,
+                                                             simdSize);
+                            return gtNewSimdAsHWIntrinsicNode(retType, tmp, NI_Vector64_ToScalar, simdBaseJitType,
+                                                              simdSize);
+                        }
```

Comment:
it would probably be good to do:
```cpp
case TYP_LONG:
case TYP_ULONG:
case TYP_FLOAT:
case TYP_DOUBLE:
{
    break;
}

default:
{
    unreached();
}
```

We may support more types in the future (such as `Half`) and so asserting for the unhandled cases tends to be "good".

Reply:
Got it. Will fix it.